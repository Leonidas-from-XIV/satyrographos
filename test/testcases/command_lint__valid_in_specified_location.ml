open Core
open Satyrographos_testlib

let satysfi_package_opam =
  "satysfi-package.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package"
    ~version:"0.1"
    ()

let satysfi_package_doc_opam =
  "satysfi-package-doc.opam", TestLib.opam_file_for_test
    ~name:"satysfi-package-doc"
    ~version:"0.1"
    ~depends:{|
  "satysfi" {>= "0.0.5" & < "0.0.6"}
  "satyrographos" {>= "0.0.2.6" & < "0.0.3"}

  "satysfi-package" {= "0.1"}
|}
    ()

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ())
  (opam "satysfi-package.opam")
  (dependencies ()))

(libraryDoc
  (name "package-doc")
  (version "0.1")
  (build ())
  (sources ())
  (opam "satysfi-package-doc.opam")
  (dependencies ((package ()))))
|}

let files =
  [ satysfi_package_opam;
    satysfi_package_doc_opam;
    satyristes;
  ]

let opam_libs = Satyrographos.Library.[
    {empty with name = Some "package"};
  ]

let () =
  let f cwd =
    ".", Some (FilePath.concat cwd "Satyristes")
  in
  TestCommand.test_lint_command ~f ~opam_libs files
