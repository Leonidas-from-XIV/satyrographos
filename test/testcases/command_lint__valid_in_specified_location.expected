0 problem(s) found.
